lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -239,28 +241,39 @@ def set_target_dependencies
               aggregate_target.native_target.add_dependency(pod_target.native_target)
               configure_app_extension_api_only_for_target(pod_target) if is_app_extension
 
-              add_dependent_targets_to_native_target(pod_target.dependent_targets,
-                                                     pod_target.native_target, is_app_extension,
-                                                     pod_target.requires_frameworks? && !pod_target.static_framework?,
-                                                     frameworks_group)
+              if is_app_extension
+                pod_extension_targets << pod_target
```

Comment:
kind of surprised this works, I was under the impression `pod_target` does not implement equality

Reply:
That means it gets the default implementation, which is identity for equality and object id for hash value